virttest/test_setup.py

```diff
@@ -304,21 +304,20 @@ def __init__(self, params):
         self.pool_path = "/sys/kernel/mm/hugepages"
         self.sys_node_path = "/sys/devices/system/node"
         # Unit is KB as default for hugepage size.
-        default_hugepage_size = 2048
         try:
             self.expected_hugepage_size = int(
-                params.get("expected_hugepage_size", default_hugepage_size))
+                params.get("expected_hugepage_size", 0))
         except TypeError:
-            logging.warn("Using default value %s as "
-                         "'expected_hugepage_size=%s' is invalid",
-                         default_hugepage_size,
+            logging.warn("Invalid value 'expected_hugepage_size=%s'",
                          params.get("expected_hugepage_size"))
-            self.expected_hugepage_size = default_hugepage_size
+            self.expected_hugepage_size = 0
         self.hugepage_cpu_flag = params.get("hugepage_cpu_flag")
         self.hugepage_match_str = params.get("hugepage_match_str")
-        self.check_hugepage_support()
+        if self.hugepage_cpu_flag and self.hugepage_match_str:
+            self.check_hugepage_support()
```

Comment:
Let's move this check in check_hugepage_support().  Because 'hugepage_match_str'  is a optional param for 2M  hugepage, but cpu feature 'pse' is required. 
For 1G hugepage on x86 platform, cpu feature 'pdpe1gb'  and kernel param 'hugepagesz=1GB' are required.